dwcj-engine/src/main/java/org/dwcj/component/label/Label.java

```diff
@@ -3,59 +3,237 @@
 import com.basis.bbj.proxies.sysgui.BBjStaticText;
 import com.basis.bbj.proxies.sysgui.BBjWindow;
 import com.basis.startup.type.BBjException;
-
 import org.dwcj.Environment;
 import org.dwcj.bridge.WindowAccessor;
 import org.dwcj.component.AbstractDwcComponent;
 import org.dwcj.component.TextAlignable;
+import org.dwcj.component.event.EventDispatcher;
+import org.dwcj.component.event.EventListener;
+import org.dwcj.component.event.MouseEnterEvent;
+import org.dwcj.component.event.MouseExitEvent;
+import org.dwcj.component.event.RightMouseDownEvent;
+import org.dwcj.component.event.sink.MouseEnterEventSink;
+import org.dwcj.component.event.sink.MouseExitEventSink;
+import org.dwcj.component.event.sink.RightMouseDownEventSink;
 import org.dwcj.component.window.AbstractWindow;
 import org.dwcj.util.BBjFunctionalityHelper;
 
+/** A label object. */
 public final class Label extends AbstractDwcComponent implements TextAlignable {
 
-  /*
-   * top, right, bottom, left margins, used simple array as user will not have to interact directly
-   * with this
-   */
+  private EventDispatcher dispatcher = new EventDispatcher();
+  private MouseEnterEventSink mouseEnterEventSink;
+  private MouseExitEventSink mouseExitEventSink;
+  private RightMouseDownEventSink rightMouseDownEventSink;
+  private boolean lineWrap = true;
+  private BBjStaticText component;
 
   public Label() {
     this("");
   }
 
   /**
-   * Constructor used to give the label initial text
+   * Constructor used to give the label initial text.
    *
    * @param text String value for initial display text
    */
   public Label(String text) {
     setText(text);
-    this.textAlignment = Alignment.LEFT;
   }
 
+  /**
+   * Constructor used to give the label wether it is linewrapped or not.
+   *
+   * @param wrap Boolean value for linewrapping.
+   */
+  public Label(boolean wrap) {
```

Comment:
The constructor should accept both text and wrap

Reply:
I added two constructors, one for just wrap which will use default empty string for text and one for both.